ProductFunctions/ProductFunctions/ProductFunctions.cs

```diff
@@ -0,0 +1,96 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.IO;
+using System;
+using System.Web.Http;
+
+namespace ProductFunctions
+{
+    public static class ProductFunctions
+    {
+
+        static DummyDB dummyDB = new DummyDB();
+
+        [FunctionName("getAllProducts")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", Route = "products")] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation("C# HTTP trigger function processed a request.");
+
+            var existingProducts = JsonConvert.SerializeObject(dummyDB.onDisplayProducts, Formatting.Indented);
```

Comment:
Move all "DB" logic to a separate file
Create folder services and add there the class ProductService. This should have 4 methods (for GET, POST, PUT, DELETE)
On ProductFunctions use ProductService methods.
In general It's better to separate the logic as much as possible. This way, when you will implement the actual db connection you won't have to change the implementation in ProductFunctions, you will just have to implement the 4 methods with the DB connection.